src/containers/Dhis2DataElementsProvider.js

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import Api from "../lib/Api";
+import { useStore } from "react-redux";
+import { dhis2ReceiveDe, dhis2RequestDe } from "../actions/dhis2";
+import keyBy from "lodash/keyBy";
+
+/*
+  here to populate the store with data elements and indicator cache for faster display within the children
+  see ../lib/dhis2lookups{*} for pratical usage
+*/
+const Dhis2DataElementsProvider = props => {
+  const dispatch = useDispatch();
+  const store = useStore();
+  useEffect(() => {
+    const fetchDataElements = async () => {
+      if (store.getState().dhis2.status === null) {
+        dispatch(dhis2RequestDe());
+        const api = await Api.instance();
+        const deResp = await api.get("dataElements", {
```

Comment:
I might have moved the api part into a seperate function that returns dataelements, indicators, cocs. Because they don't really need the store.